File: App_Code/BudgetItem.cs
public class BudgetItem
{
    public string name { get; set; }
    public int type { get; set; }
    public float currency { get; set; }
    private Period _period = new Period();

    public Period period
    {
        get
        {
            return _period;
        }

        set
        {
            _period = value;
        }
    }
}

File: Scripts/budgetItem.js
app.factory('budgetItem', ['period', function budgetItemFactory(period) {
    return function(name, type){
        return {
            name: name,
            type: type,
            currency: undefined,
            period: period.setPeriod(0, 0, 1, 31, 0, 11, new Date())
        }
    }
}]);